galasa-ui/src/app/mysettings/page.tsx

```diff
@@ -0,0 +1,156 @@
+                newSet.add(tokenId);
+                return newSet;
+            });
+
+        }
+    }
+
+    const handleFetchAccessTokens = async () => {
+
+        setIsLoading(true)
+
+        try {
+            const response = await fetch('/auth/tokens', { method: 'GET' })
+
+            if (response.ok) {
+
+                const data = await response.json();
+
+                setTokens(data)
+
+            }
+
+        } catch (err) {
+            setIsError(true)
+        }
+        finally {
+            setIsLoading(false)
+        }
+
+    }
+
+    const handleUpdateDeleteModalState = () => {
+        setIsDeleteModalOpen(false)
+    }
+
+    const handleUpdateTokens = (token : any) => {
+        setTokens((prevTokens) => {
+            const newTokens = new Set(prevTokens);
+            newTokens.delete(token);
+            return newTokens;
+        });
+
+        setSelectedTokens((prevSelectedTokens) => {
+            const newSelectedTokens = new Set(prevSelectedTokens);
+            newSelectedTokens.delete(token.tokenId);
+            return newSelectedTokens;
+        });
+
+        setIsDeleteModalOpen(false)
+    }
+
+    useEffect(() => {
+
+        handleFetchAccessTokens()
+
+    }, [])
+
+    if (isLoading) {
+        return <Loading />
+    }
+
+    if (isError) {
+        return (
+            <ErrorPage />
+        )
+    }
+
+    return (
+        <div id="content" className={styles.content}>
+            <PageTile title={"My Settings"} />
+
+            <div className={styles.tokenContainer}>
+                <h4 style={{ margin: "1rem 0" }}>Access Tokens</h4>
+
+                <div className={styles.titleContainer}>
+                    <div>
+                        <h5 className={styles.heading}>An access token is a unique secret key held by a client program so it has permission to use the Galasa service</h5>
+                        <h5 className={styles.heading}>A token has the same access rights as the user who allocated it.</h5>
+                    </div>
+
+                    <div>
+                        <TokenRequestModal isDisabled={selectedTokens.size > 0 ? true : false} />
+
+                        <Button onClick={() => setIsDeleteModalOpen(true)} className={styles.deleteBtn} disabled={selectedTokens.size === 0} kind="danger">
+                            Delete {selectedTokens.size} selected access tokens
+                        </Button>
+
+                        <TokenResponseModal/>
+
+                    </div>
+                </div>
+
+                <div title="Access Tokens" className={styles.tokensList}>
+                    {
+                        Array.from(tokens).map((token) => (
+                            <TokenCard key={token.tokenId} tokenId={token.tokenId} tokenDescription={token.description} createdAt={token.creationTime} owner={token.owner.loginId} handleSelectTokensForDeletion={handleSelectTokensForDeletion} />
```

Comment:
We can just pass the token as a property into the TokenCard component and that can then pull out what it needs rather than doing it here